container/container.go

```diff
@@ -463,6 +465,9 @@ func AttachStreams(ctx context.Context, streamConfig *runconfig.StreamConfig, op
 			}
 			<-done
 		}
+		if detached {
```

Comment:
Given that only daemon can log an event, `AttachStream` can't log an event directly.  So at least we can't `find an error, spawn event, and return error` directly. I think there're some choices:
* let AttachStream return multiple errors via error channel, first is detach error, others are unexpected errors.
* use additional detach channel as previous implementation. 

What do you suggest?